napari/_qt/widgets/qt_dask_settings.py

```diff
@@ -0,0 +1,315 @@
+from qtpy.QtCore import Signal
+from qtpy.QtWidgets import QCheckBox, QHBoxLayout, QLabel, QSpinBox, QWidget
+
+from ...utils.translations import trans
+
+
+class QtDaskSettingsWidget(QWidget):
+
+    """Creates custom widget to set enable/disable dask and set cache size.
+
+    Parameters
+    ----------
+    description : str
+        Text to explain and display on widget.
+    enabled: bool
+        Bool value indicating if dask is enabled.
+    value : dict
+        value = {'enabled': dask enabled (True/False),
+        'cache': cache size in mb}
+    cache : int
+        Dask cache size in mb.
+    min_value : int
+        Minimum value of allowable cache range.
+    max_value : int
+        Maximum value of allowable cache range.
+    inc : int
+        Increment of cache step for cache value widget (in mb).
```

Comment:
One example of simplification: perhaps we get rid of inc (value can only be set by typing), and we leave the range checking to the base model rather than the UI? e.g. I can type "5000000" in the box, but when Qt tries to update the model, it hits the max, and that updates the value to the max. (Perhaps raising a warning, though I can live without it.)

Reply:
I believe that this is resolved by the widget refactoring.